exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/visitor/JoinPrelRenameVisitor.java

```diff
@@ -68,4 +69,27 @@ public Prel visitJoin(JoinPrel prel, Void value) throws RuntimeException {
     return (Prel) prel.copy(prel.getTraitSet(), reNamedChildren);
   }
 
+  @Override
+  public Prel visitCorrelate(CorrelatePrel prel, Void value) throws RuntimeException {
+
+    List<RelNode> children = Lists.newArrayList();
+
+    for(Prel child : prel){
+      child = child.accept(this, null);
+      children.add(child);
+    }
+
+    final int leftCount = children.get(0).getRowType().getFieldCount();
+
+    List<RelNode> reNamedChildren = Lists.newArrayList();
+
+    RelNode left = prel.getCorrelateInput(0, children.get(0));
```

Comment:
Yes. We wanted to make renaming only on right side. When I looked at the code it looks like all the time it should modify only the right side, because it first checks the left side and then the right side. Hence I thought same logic should work. Please let me know if I am missing anything here.